README.md

```diff
@@ -4,183 +4,200 @@
-values for each service. You may also specify `weight` and/or `host`.  They each
-default to a zero value. Please consult the RFC for an explanation of these
-values. You can also specify `txt_payload` which is used to define the data in
-a TXT DNS resource record, it should be a list of strings containing a key and
-value separated by a `=`. Services can be configured in `config.exs` as shown
-above, or at runtime:
+Here's how to add and remove a service at runtime:
 
 ```elixir
-iex> services = [
-  # service type: _http._tcp.local - used in match
-  %{
-    name: "Web Server",
+iex> MdnsLite.add_mdns_service(%{
+    id: :my_web_server,
     protocol: "http",
     transport: "tcp",
     port: 80,
-  },
-  # service_type: _ssh._tcp.local - used in match
-  %{
-    name: "Secure Socket",
-    protocol: "ssh",
-    transport: "tcp",
-    port: 22,
-    txt_payload: ["key=value"],
-  }
-]
-
-iex> MdnsLite.add_mds_services(services)
+  })
+:ok
+iex> MdnsLite.remove_mdns_service(:my_web_server)
 :ok
 ```
 
-Services can also be removed at runtime via `remove_mdns_services/1` with the
-service name to remove:
+## Client
 
-```elixir
-iex> service_names = ["Web Server", "Secure Socket"]
-iex> MdnsLite.remove_mdns_services(services)
-:ok
+`MdnsLite.gethostbyname/1` uses mDNS to resolve hostnames. Here's an example:
 
-# Remove just a single service
-iex> MdnsLite.remove_mdns_services("Secure Socket")
-:ok
+```elixir
+iex> MdnsLite.gethostbyname("my-laptop.local")
+{:ok, {172, 31, 112, 98}}
 ```
 
-## Installation
+If you just want mDNS to "just work" with Erlang, you'll need to enable
+MdnsLite's DNS Bridge feature and configure Erlang's DNS resolver to use it. See
+the DNS Bridge section for details.
 
-If [available in Hex](https://hex.pm/docs/publish), the package can be installed
-by adding `mdns_lite` to your list of dependencies in `mix.exs`:
+Service discovery docs TBD...
+
+## DNS Bridge configuration
+
+`MdnsLite` can start a DNS server to respond to `.local` queries. This enables
+code that has no knowledge of mDNS to resolve mDNS queries. For example,
+Erlang/OTP's built-in DNS resolver doesn't know about mDNS. It's used to resolve
+hosts for Erlang distribution and pretty much any code using `:gen_tcp` and
+`:gen_udp`. `MdnsLite`'s DNS bridge feature makes `.local` hostname lookups work
+for all of this. No code modifications required.
+
+Note that this feature is useful on Nerves devices. Erlang/OTP can use the
+system name resolver on desktop Linux and MacOS. The system name resolver should
+already be hooked up to an mDNS resolver there.
+
+To set this up, you'll need to enable the DNS bridge on `MdnsLite` and then set
+up the DNS resolver to use it first. Here are the options for the application
+environment:
 
 ```elixir
-def deps do
-  [
-    {:mdns_lite, "~> 0.6"}
-  ]
-end
+config :mdns_lite,
+  dns_bridge_enabled: true,
+  dns_bridge_recursive: true
+
+config :vintage_net,
+  additional_name_servers: [{127, 0, 0, 53}]
```

Comment:
Why is this needed and whats the logic for a `.53` address?

Reply:
It could be anything on 127.0.0.0/8. I picked 127.0.0.53 for a few reasons:

1. My first try was to just support a random port number, and I have partial support for that in `:vintage_net`, but then I found out that Erlang doesn't support non-port 53 specs in `resolv.conf`.
2. Second try was 127.0.0.1:53, but I didn't like that since it seemed some someone might run a DNS server on a Nerves device and it seemed better to have that port available.
3. The next try was picking a random local address. 127.0.0.53 seemed as good as any and that was the choice made by the Systemd people as well, so I did that.

Reply:
I just added a little more text to the README on this.